### PR TITLE
Clone step functionality

### PR DESCRIPTION
**Pull request by matt-richardson**
Adds the ability to clone a (non-child) step to the same project.

### CODE REVIEW

**Comment by matt-richardson** [comment]
Hmmm... Build faild. Passes locally though:

```
ConsoleReporter is deprecated and will be removed in a future version.Started
..........................

26 specs, 0 failures
Finished in 0.036 seconds
```

**Comment by matt-richardson** [comment]
Updated to work under both versions of phantom, and under Chrome 47.0.2526.80 m

**Comment by davidroberts63** [comment]
Awesome! Haven't looked at the code (your commit messages are great) but definitely will do. And thanks for getting it working under 1.9.8. I need to update the build to use 2.0 sometime soon. Thanks!

**Comment by davidroberts63** [comment]
Code looks good. I'll test it on my installation tonight.

File: src/3.0/clone-step.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            attribute.value = "";
+            span.setAttributeNode(attribute);
+
+            document.body.appendChild(span);
+
+            console.debug(" - creating refresh handler script")
+            var script = document.createElement("script");
+            script.id = 'bluefin-clonestep-refreshhandler';
+            script.type = 'text/javascript';
+            script.text = "var button = document.querySelector('#bluefin-clonestep-refreshbutton'); button.onClick = button.onclick = function() { if (this.attributes['status'].value == 'success') { angular.element(\"#processEditDropdown\").injector().get(\"$route\").reload();} else { angular.element(\"#processEditDropdown\").injector().get(\"octoDialog\").messageBox('Clone Step Failed', this.attributes['message'].value, [{label: 'ok'}]);} }";
```
Comment (davidroberts63) [inline src/3.0/clone-step.js]: Interesting, never thought of injecting a script element to gain access to the site's angular.

Comment (matt-richardson) [inline src/3.0/clone-step.js]: Banged my head against the wall quite a bit before I got that working...

File: src/3.0/clone-step.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    function addCloneStepMenuItem() {
+        console.debug("Adding clone menu item to dropdown menu")
+        var menu = document.querySelector("ul.dropdown-menu[role='menu']");
+
+        var newMenuItem = document.createElement('li')
+        newMenuItem.className = 'divider';
+        menu.appendChild(newMenuItem);
+
+        newMenuItem = document.createElement('li')
+        newMenuItem.innerHTML = '<a tabindex="-1">Clone</a>'
```
Comment (davidroberts63) [inline src/3.0/clone-step.js]: Add an empty href attribute "<a tabindex="-1" href>Clone</a>" so that it will have the hand icon on it like the others.

**Comment by davidroberts63** [comment]
Looks wonderful. Test's out perfect. I really like this feature. I'll publish it tomorrow evening.

File: src/3.0/clone-step.js (hunk 2 above)
Comment (matt-richardson) [inline src/3.0/clone-step.js]: Done.

**Comment by davidroberts63** [comment]
Still checking something on this. After I merged it I'm seeing an odd javascript error after cloning coming from that script embeding. It has to be something I did because it was working before I merged.

**Comment by matt-richardson** [comment]
Odd. I've pulled the 2 branches, and they look identical. Can you share the error, and repro?

**Comment by matt-richardson** [comment]
Figured it. It was the adding of the `href=""` part. Causes a navigate. 
Will raise a new PR.

**Comment by davidroberts63** [comment]
Perfect. You're awesome. Really, this whole feature is a wonderful PR. I'll try it again this evening and hopefully publish then.

**Pull request merged**